xbmc/utils/MovingSpeed.h

```diff
@@ -0,0 +1,77 @@
+/*
+ *  Copyright (C) 2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include <map>
+#include <stdint.h>
+
+namespace UTILS
+{
+namespace MOVING_SPEED
+{
+
+struct EventCfg
+{
+  friend class CMovingSpeed;
+  /*!
+   * \param acceleration Acceleration in pixels per second (px/sec)
+   * \param maxVelocity Max movement speed, it depends from acceleration value,
+   *        a suitable value could be (acceleration value)*3. Set 0 to disable it
+   * \param resetTimeout Resets acceleration speed if idle for specified millisecs
+   */
+  EventCfg(float acceleration, float maxVelocity, uint32_t resetTimeout)
+    : m_acceleration{acceleration}, m_maxVelocity{maxVelocity}, m_resetTimeout{resetTimeout}
+  {
+  }
+
+  float m_acceleration;
+  float m_maxVelocity;
+  uint32_t m_resetTimeout;
+
+private:
+  float m_currentVelocity = 1.0f;
+  uint32_t m_lastFrameTime = 0;
+};
+
+class CMovingSpeed
```

Comment:
I like that you encapsulated the moving speed in this new class. The class could use a really top-quality doxy string that scopes what use case the new class is solving, and some description of the desired motion effect.

Reply:
add it i hope no make writing mistakes

Reply:
looks great!